benchcab/config.py

```diff
@@ -142,8 +151,8 @@ def read_config(config_path: str) -> dict:
     """
     # Read configuration file
     config = read_config_file(config_path)
-    # Populate configuration dict with optional keys
-    read_optional_key(config)
     # Validate and return.
     validate_config(config)
+    # Populate configuration dict with optional keys
+    read_optional_key(config)
```

Comment:
Why did you swap the ordering of the calls?

Reply:
Since there are different requirements for validating config before vs. after adding in the optional keys (setting  (`required` -> `true/false`, `nullable` -> `true`) for optional keys. Since we want a self-documenting nature for `config-schema.yml` with respect to the user input, and the behaviour of `validate_config` is checked by unit tests, keeping the original order would be fine as suggested.